runtime/src/main/java/org/corfudb/runtime/collections/TxnContext.java

```diff
@@ -102,10 +105,24 @@ void validateTableWrittenIsInNamespace(@Nonnull Table<K, V, M> table) {
                     "TxnContext cannot be used after a transaction has ended on "+
                     table.getFullyQualifiedTableName());
         }
+        if (validateKey && key == null) {
+            throw new IllegalArgumentException("Key cannot be null on "
+                    + table.getFullyQualifiedTableName() + " in transaction on namespace " + namespace);
+        }
```

Comment:
Seems a little weird to manipulate the bitmask in a validate method.

Reply:
sure, open to any alternate suggestions for a better name.
How about `preProcessMutation()` ?